### PR TITLE
add command to summarize space usage

### PR DESCRIPTION
**Pull request by icewind1991**
Helps with figuring out what is taking up the space or debugging quota issues.

example output:

```
files: 9.6 MB
- src: 8.6 MB
  - private: 4.1 MB
    - Files: 952 KB
      - Storage: 219 KB
        - Wrapper: 104 KB
      - Cache: 196 KB
      - ObjectStore: 95 KB
      - Node: 72 KB
    - DB: 243 KB
      - QueryBuilder: 120 KB
    - AppFramework: 238 KB
    - Share20: 221 KB
    - legacy: 205 KB
    - Authentication: 186 KB
    - User: 127 KB
    - Repair: 111 KB
    - Collaboration: 107 KB
    - Security: 107 KB
    - Preview: 98 KB
    - App: 87 KB
  - l10n: 2.5 MB
  - public: 1.5 MB
  - composer: 424 KB
- picture.png: 960 KB
```

### CODE REVIEW

**Review by come-nc — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Comment by blizzz** [comment]
I am not merging due to an open question ^

File: core/Command/Info/FileUtils.php [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	 * @param IUser $user
+	 * @return IMountPoint[]
+	 */
+	public function getMountsForUser(IUser $user): array {
+		$this->setupManager->setupForUser($user);
+		$prefix = "/" . $user->getUID();
+		return array_filter($this->mountManager->getAll(), function (IMountPoint $mount) use ($prefix) {
+			return str_starts_with($mount->getMountPoint(), $prefix);
+		});
+	}
```
[the remark this answers is not in the document]
Comment (icewind1991) [reply core/Command/Info/FileUtils.php:195]: it's used by a follow up PR this was extracted from, moving it to that PR

**Review by artonge — APPROVED** [review]

**Pull request merged**